### ISSUE TITLE
OSError: [Errno 5] Input/output error

### ISSUE THREAD

**Issue by MrBr-github** [MrBr-github/lshca#32]
```
root@swx-snake-05:~# python3 /hpc/local/bin/lshca_future
Traceback (most recent call last):
  File "/hpc/local/bin/lshca_future", line 1758, in <module>
    main()
  File "/hpc/local/bin/lshca_future", line 1752, in main
    hca_manager.get_data()
  File "/hpc/local/bin/lshca_future", line 357, in get_data
    bdf_dev.get_data()
  File "/hpc/local/bin/lshca_future", line 1205, in get_data
    self._sysFSDevice.get_data()
  File "/hpc/local/bin/lshca_future", line 881, in get_data
    self.lnk_state = self._data_source.read_file_if_exists(self._sys_prefix + "/infiniband/" + self.rdma + "/ports/" +
  File "/hpc/local/bin/lshca_future", line 1632, in read_file_if_exists
    output = f.read()
OSError: [Errno 5] Input/output error
```

```
root@swx-snake-05:~# ll /sys/bus/pci/devices/0000:86:00.0/infiniband/mlx5_0/ports/1/state
-r--r--r-- 1 root root 4096 Mar 25 09:02 /sys/bus/pci/devices/0000:86:00.0/infiniband/mlx5_0/ports/1/state
root@swx-snake-05:~# cat /sys/bus/pci/devices/0000:86:00.0/infiniband/mlx5_0/ports/1/state
cat: '/sys/bus/pci/devices/0000:86:00.0/infiniband/mlx5_0/ports/1/state': Input/output error
```

**Comment by MrBr-github** [comment]
```
root@swx-snake-05:~# cat /etc/os-release
NAME="Ubuntu"
VERSION="20.04.1 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.1 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal



root@swx-snake-05:~# modinfo mlx5_core
filename:       /lib/modules/5.4.0-48-generic/kernel/drivers/net/ethernet/mellanox/mlx5/core/mlx5_core.ko
version:        5.0-0
```

**Comment by MrBr-github** [comment]
Fixed by c912ae9c85ff20c525bcde0928d6970ef63289fb
